clash-prelude/src/Clash/Signal/Internal.hs

```diff
@@ -382,6 +412,20 @@ type DomainResetKind (dom :: Domain) =
 type DomainInitBehavior (dom :: Domain) =
   DomainConfigurationInitBehavior (KnownConf dom)
 
+-- | Convenience type to constrain a domain to have initial values. Example
+-- usage:
+--
+-- @
+-- myFunc :: HasDefinedInitialValues dom => ...
+-- @
+--
+-- Using this type implies 'KnownDomain'.
+--
+-- __N.B.__: There is no @UnknownInitialValues dom@ as a component that works
+--           without initial values will also work if it does have them.
```

Comment:
Is this **NB:** or simply "Note:"?

Do you really have to nota this that bene? ;-) NB means "Pay attention please this is important". I think this suffices:
```suggestion
-- Note that there is no @UnknownInitialValues dom@ as a component that works
-- without initial values will also work if it does have them.
```

(Also, in the majority of cases, we don't insert periods in NB. I should probably add the non-conforming cases to a doc polishing PR, but there are more pressing issues first)

Reply:
Let's get rid of it, I don't like NB anyway.